include/topgg/webhook.h

```diff
@@ -0,0 +1,119 @@
+#pragma once
+
+#include <dpp/dpp.h>
+#include <nlohmann/json.hpp>
+#include <functional>
+#include <string>
+
+namespace topgg {
+
+/**
+ * @brief Data received from a bot vote webhook
+ */
+struct bot_vote_data {
+    dpp::snowflake user_id;          ///< ID of the user who voted
+    std::string type;                ///< Type of vote (usually "upvote")
+    bool is_weekend;                 ///< Whether the vote was done during weekend
+    std::string query;               ///< Query string data if any
+    
+    static bot_vote_data from_json(const nlohmann::json& j) {
```

Comment:
This function body should be moved to `webhook.cpp` as it does not depend on any generics.